internal/httpx/resty.go

```diff
@@ -0,0 +1,60 @@
+package httpx
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/auth"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+)
+
+// Resty provides a simplified facade over the resty.Client for making HTTP requests with
+// contextual authentication headers. This facade is designed to handle client configuration,
+// setup, and authentication, abstracting away the underlying complexities of the resty library.
+type Resty struct {
+	cli *resty.Client
+}
+
+// Get performs an HTTP GET request to the specified path using the configured resty.Client.
+// It attaches authentication headers and handles errors gracefully, returning the response body
+// or an error if the request fails.
+func (r *Resty) Get(ctx context.Context, path string) ([]byte, error) {
+	req := r.cli.R().SetContext(ctx)
+	res, err := req.Get(path)
+	if err != nil {
+		return nil, fmt.Errorf("HTTP GET %s: %w", r.cli.BaseURL+path, err)
+	}
+	return res.Body(), nil
+}
+
+// NewResty creates and configures a Resty instance for authenticated requests.
+func NewResty(addr string, cfg *auth.HeaderConfig) (*Resty, error) {
+	b, err := auth.NewHeaderBuilder(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build HTTP auth header: %w", err)
+
+	}
+	cli := resty.New().
+		SetError(&models.SPVError{}).
+		SetBaseURL(addr).
+		OnBeforeRequest(authHeaderMiddleware(b))
```

Comment:
Consider setting timeout value.

Reply:
1. Timeouts can be set by passing the context with timeout to the function/method by the caller. 
2. However, if somebody decides to make a call without such context type, we can set the default timeout for the client.

Reply:
1. Yes. But `resty` will nontheless add it's own timeout - and we should use/control it. 
2. For now, every "user" of this package uses it with standard `context.Background` - so the default timeout is necessary.

Reply:
I added the configuration for the client where you can set the necessary timeout value during initialization. In case of the local development there is a DefaultConfig.